WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java

```diff
@@ -0,0 +1,590 @@
+    private StockMediaRetainedFragment mRetainedFragment;
+
+    private RecyclerView mRecycler;
+    private ViewGroup mSelectionBar;
+    private TextView mTextAdd;
+
+    private SearchView mSearchView;
+    private String mSearchQuery;
+    private final Handler mHandler = new Handler();
+
+    private int mThumbWidth;
+    private int mThumbHeight;
+
+    private boolean mIsFetching;
+    private boolean mCanLoadMore;
+    private int mNextPage;
+
+    @Inject Dispatcher mDispatcher;
+    @Inject StockMediaStore mStockMediaStore;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ((WordPress) getApplication()).component().inject(this);
+        setContentView(R.layout.stock_media_picker_activity);
+
+        if (savedInstanceState == null) {
+            mSite = (SiteModel) getIntent().getSerializableExtra(WordPress.SITE);
+        } else {
+            mSite = (SiteModel) savedInstanceState.getSerializable(WordPress.SITE);
+        }
+        if (mSite == null) {
+            ToastUtils.showToast(this, R.string.blog_not_found, ToastUtils.Duration.SHORT);
+            finish();
+            return;
+        }
+
+        FragmentManager fm = getFragmentManager();
+        mRetainedFragment = (StockMediaRetainedFragment) fm.findFragmentByTag(TAG_RETAINED_FRAGMENT);
+        if (mRetainedFragment == null) {
+            mRetainedFragment = StockMediaRetainedFragment.newInstance();
+            fm.beginTransaction().add(mRetainedFragment, TAG_RETAINED_FRAGMENT).commit();
+        }
+
+        int displayWidth = DisplayUtils.getDisplayPixelWidth(this);
+        mThumbWidth = displayWidth / getColumnCount();
+        mThumbHeight = (int) (mThumbWidth * 0.75f);
+
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayHomeAsUpEnabled(true);
+            actionBar.setDisplayShowTitleEnabled(true);
+        }
+
+        mRecycler = findViewById(R.id.recycler);
+        mRecycler.setLayoutManager(new GridLayoutManager(this, getColumnCount()));
+
+        mAdapter = new StockMediaAdapter();
+        mRecycler.setAdapter(mAdapter);
+
+        mSelectionBar = findViewById(R.id.container_selection_bar);
+        mTextAdd = findViewById(R.id.text_add);
+        mTextAdd.setOnClickListener(new View.OnClickListener() {
+            @Override public void onClick(View v) {
+                uploadSelection();
+            }
+        });
+        findViewById(R.id.text_clear).setOnClickListener(new View.OnClickListener() {
+            @Override public void onClick(View v) {
+                mAdapter.clearSelection();
+            }
+        });
+
+        if (savedInstanceState == null) {
+            showEmptyView(true);
+        } else {
+            mSearchQuery = savedInstanceState.getString(KEY_SEARCH_QUERY);
```

Comment:
I just want to say that I really hate how inconsistent Android is when it comes to whether the UI component will retain its data or not. `SearchView` does retain it, so there is nothing we need to do here, but I still hate it because I can't remember which ones do and have to check it each time 🤦‍♂️

Reply:
Completely agree, especially since `SearchView` submits the retained query without you having any control over it.